src/java/org/apache/cassandra/service/accord/AccordJournal.java

```diff
@@ -1123,17 +1111,17 @@ private void doRun()
                 }
                 else
                 {
-                    requestBuffer.add(request);
+                    if (null == requests) requests = new ArrayList<>();
```

Comment:
Hard to say without knowing the actual inputs for this in practice, but `requests` might actually make more sense to have a fixed buffer. One thing you might want to do is artificially constrain frame size to some cache friendly number. I'll bet allocate and copy is plenty fast in that situation, and faster than doing even a few resizes.

But if this is always pretty small then it doesn't help much, but also doesn't hurt?

Reply:
We might indeed. I'll make a note, and possibly revisit in the future, if that's okay.